File: test/octave/mRBDL/Tools/PlotModel.m
function PlotModel(model, q)
%PlotModel - Plot links of model
%
% Syntax: PlotModel(q)
%
% Long description
pos_o = [];
pos_e = [];
num = max(size(model.idlinkplot,1), size(model.idlinkplot,2));
for i=1:num
    pos_o(:,i) = CalcBodyToBaseCoordinates(model, q, model.idlinkplot(i), zeros(3,1));
    pos_e(:,i) = CalcBodyToBaseCoordinates(model, q, model.idlinkplot(i), model.linkplot{i});
end

nc = max(size(model.idcontact,1), size(model.idcontact,2));
for i=1:nc
    pos_contact(:,i) = CalcBodyToBaseCoordinates(model, q, model.idcontact(i), model.contactpoint{i});
end

PlotLink(pos_o, pos_e, num, pos_contact);
    
end

File: src/jaxRBDL/Tools/PlotModel.py
from matplotlib.pyplot import axis
from mpl_toolkits.mplot3d.axes3d import Axes3D
from numpy.testing._private.utils import import_nose
from matplotlib.figure import Figure
import matplotlib.pyplot as plt
import numpy as np
from jaxRBDL.Kinematics.CalcBodyToBaseCoordinates import CalcBodyToBaseCoordinates
from jaxRBDL.Tools.PlotLink import PlotLink

def PlotModel(model: dict, q: np.ndarray, ax: Axes3D):

    idlinkplot = model["idlinkplot"]
    linkplot = model["linkplot"]
    idcontact = model["idcontact"]
    contactpoint = model["contactpoint"]

    pos_o = []
    pos_e = []

    num = len(idlinkplot)

    for i in range(num):
        pos_o.append(CalcBodyToBaseCoordinates(model, q, idlinkplot[i], np.zeros((3,1))))
        pos_e.append(CalcBodyToBaseCoordinates(model, q, idlinkplot[i], linkplot[i]))

    pos_o = np.concatenate(pos_o, axis=1)
    pos_e = np.concatenate(pos_e, axis=1)

    nc = len(idcontact)

    pos_contact = []
    for i in range(nc):
        pos_contact.append(CalcBodyToBaseCoordinates(model, q, idcontact[i], contactpoint[i]))
    pos_contact = np.concatenate(pos_contact, axis=1)
    
    ax = PlotLink(pos_o, pos_e, num, pos_contact, ax)
    # ax = PlotLink(pos_o, pos_e, 16, pos_contact, ax)
    return ax

